tracer/span.go

```diff
@@ -136,7 +152,20 @@ func (s *Span) SetError(err error) {
 		return
 	}
 
+	// FIXME: We acquire and release the lock several times over the course of
```

Comment:
At some point you should do a pass where you replace the span's lock with an `RWMutex`, and redo the way locking works so you only need to acquire the lock once in this function. I think it's out of scope to fix that in this PR.